guides/tutorials/nextjs-openai-chatbot.mdx

```diff
@@ -0,0 +1,592 @@
+  }: {                                                         // <-
+    messageId: string;                                         // <-
+    stepId: string;                                            // <-
+    upvote: boolean;                                           // <-
+  }) => {                                                      // <-
+    fetch("api/score", {                                       // <-
+      method: "POST",                                          // <-
+      headers: { "Content-Type": "application/json" },         // <-
+      body: JSON.stringify({ stepId, score: upvote ? 1 : 0 }), // <-
+    }).then(() => {                                            // <-
+      if (upvote) {                                            // <-
+        // Add the message ID to the list of upvoted messages
+        setUpvotedMessages([...upvotedMessages, messageId]);   // <-
+      }                                                        // <-
+    });                                                        // <-
+  };                                                           // <-
+  
+  return (
+	  // ...
+    {/* To avoid double scoring a chatbot's response, we will add
+      a visual cue to already upvoted messages, and disable their
+      scoring button */}
+    {!!stepId && (
+	    <button
+	      onClick={() =>
+	        handleScore({ messageId: message.id, stepId, upvote: true }) // <-
+	      }
+	      disabled={upvotedMessages.includes(message.id)} // <-
+	      className={
+	        upvotedMessages.includes(message.id) ? "" : "grayscale" // <-
+	      }
+	    >
+	      👍
+	    </button>
+	  )}
+  )
+}
+```
+
+Now you can upvote messages in your conversation, and if you navigate to the corresponding thread in Literal AI you can see that the Assistant Run has been scored :
+
+![Step with score](images/guides/nextjs-openai-chatbot/12-literal-step-with-score.png)
+
+# Conclusion
+
+Congratulations on building your own monitored chatbot! As you can see, one of the priorities of Literal AI is to make it easy to integrate with your existing codebase, while providing powerful tools to monitor and improve your AI applications. Here is a rundown of the features we have implemented in this tutorial, and how they will affect the lifecycle of your chatbot :
+
+- **System Prompt** : By decoupling the prompt from the code, you can iterate on system prompts and LLM settings, without leaving your web browser - and without having to redeploy your application !
+- **Monitoring** : By logging each conversation in a structured way, you can easily replay all the interactions between your users and your chatbot.
+- **Human Evaluation** : By allowing your users to score the chatbot’s responses, you can now passively gather a dataset of known good interactions. Because each of these interactions is linked to a specific Prompt version, you can easily evaluate the impact of prompt variations and LLM settings on the chatbot’s performance.
```

Comment:
```suggestion
- **Human Evaluation**: By allowing your users to score the chatbot’s responses, you can now passively gather a dataset of known good interactions. Because each of these interactions is linked to a specific Prompt version, you can easily evaluate the impact of prompt variations and LLM settings on the chatbot’s performance.
```